### PR TITLE
Update current usage of the term 'rating'

### PR DESCRIPTION
**Pull request by ps2611**
- Currently 'rating' is used as difference in +/- votes to list and determine
  popular reviews
- Changed it to 'net_votes' to avoid conflict with entity rating

I didn't change it to 'popularity' because we may change the criteria to determine popular reviews after the rating system has been implemented

### CODE REVIEW

**Comment by gentlecat** [comment]
> I didn't change it to 'popularity' because we may change the criteria to determine popular reviews after the rating system has been implemented

Popularity of a review would have nothing to do with a rating system though. One is determined by readers and another by authors.

**Comment by ps2611** [comment]
Agreed. I misunderstood it as popularity of entity rather than of the review.
So, I will change 'net_votes' to 'popularity'.

**Comment by gentlecat** [comment]
I made some changes to the tests, so it would be nice if you could rebase on top of the latest commit on the *master* branch. I'm also still waiting for the changes I mentioned before:

> Make sure to replace the term `rating` in other places where it's used as well. We don't want to have multiple terms describing the same thing. This will cause confusion in the future.

Tell me if you need pointers for this and when this can be done so that we are on time.

File: critiquebrainz/ws/review/views.py [hunk 1]
```diff
@@ -283,10 +283,10 @@ def review_list_handler():
                 "id": "CC BY-NC-SA 3.0",
                 "info_url": "https:\/\/creativecommons.org\/licenses\/by-nc-sa\/3.0\/"
               },
-              "rating": 0,
               "source": "BBC",
               "source_url": "http:\/\/www.bbc.co.uk\/music\/reviews\/vh54",
               "text": "REVIEW TEXT GOES HERE",
+              "rating": 100, (Rating part of the review)
```
Comment (gentlecat) [inline critiquebrainz/ws/review/views.py]: This value is not returned, so it shouldn't be there.

**Comment by gentlecat** [comment]
There are still some usages left. Here's a list:
screenshot 2017-07-14 20 39 28

File: critiquebrainz/ws/review/views.py (hunk 1 above)
Comment (ps2611) [inline critiquebrainz/ws/review/views.py]: This is what I asked about earlier. This should be done when making changes in the ws code for rating feature.

**Comment by ps2611** [comment]
The changes in db/review and db/review_test are in https://github.com/metabrainz/critiquebrainz/pull/125.
They are a part of changes in tests.

**Comment by gentlecat** [comment]
> The changes in db/review and db/review_test are in #125.They are a part of changes in tests.

Why can't they be a part of this one? I think they should be.

Think about it this way: a pull request should contain a change that can be merged and deployed on it's own. We can easily avoid confusion that having multiple terms for the same thing (or same term describing different things) can cause by changing it all here. The change is not going to be that big.

File: critiquebrainz/ws/review/views.py (hunk 1 above)
Comment (gentlecat) [inline critiquebrainz/ws/review/views.py]: Right, this is not related to renaming existing "rating" usage to "popularity".

**Comment by ps2611** [comment]
Understood. Making the required changes now
And should we change it in db/users as well because it was used previously too and its pretty isolated?
If yes, then change it to what?

File: critiquebrainz/db/review.py [hunk 2]
```diff
@@ -415,10 +415,11 @@ def list_reviews(*, inc_drafts=False, inc_hidden=False, entity_id=None,
                SUM(
                    CASE WHEN vote = 't' THEN 1
                    WHEN vote = 'f' THEN -1 WHEN vote IS NULL THEN 0 END
-               ) AS rating,
+               ) AS popularity,
                latest_revision.id as latest_revision_id,
                latest_revision.timestamp as latest_revision_timestamp,
                latest_revision.text as text,
+               latest_revision.rating as rating,
```
Comment (gentlecat) [inline critiquebrainz/db/review.py]: Also make sure to avoid making additions to SQL queries (it's fine to change names). This should be done separately when you have an actual implementation in works.

**Comment by gentlecat** [comment]
> And should we change it in db/users as well because it was used previously too and its pretty isolated?

"Rating" there is used in a completely different context. It's a rating value for images coming from Gravatar. So there's no need to change that.

File: critiquebrainz/db/review.py (hunk 2 above)
Comment (ps2611) [inline critiquebrainz/db/review.py]: I have to do it for tests otherwise they will fail.

Comment (gentlecat) [inline critiquebrainz/db/review.py]: Which tests?

Comment (ps2611) [inline critiquebrainz/db/review.py]: We have to do it now for tests otherwise they will fail.

File: critiquebrainz/ws/review/views.py [hunk 3]
```diff
@@ -283,10 +283,10 @@ def review_list_handler():
                 "id": "CC BY-NC-SA 3.0",
                 "info_url": "https:\/\/creativecommons.org\/licenses\/by-nc-sa\/3.0\/"
               },
+              "rating": 0,
```
Comment (gentlecat) [inline critiquebrainz/ws/review/views.py]: `rating` values shouldn't be listed there if they are no longer returned though. Just replace them with `popularity`.

File: critiquebrainz/db/review.py (hunk 2 above)
Comment (ps2611) [inline critiquebrainz/db/review.py]: Here https://github.com/metabrainz/critiquebrainz/pull/125/files#diff-f16a779841395de9d9362b2168bf320f

Comment (gentlecat) [inline critiquebrainz/db/review.py]: Then the pull request #125 would be the place to make that change.

File: critiquebrainz/ws/review/views.py (hunk 3 above)
Comment (ps2611) [inline critiquebrainz/ws/review/views.py]: Take a look at this:https://github.com/metabrainz/critiquebrainz/pull/125/files#diff-c15d6e193e4bf52d4a77707695d5f205
Rating will be returned in the context of entity rating. So, will have to change that appropriately when we work on ws for rating feature

Comment (gentlecat) [inline critiquebrainz/ws/review/views.py]: I understand that this will change in the future, but this has nothing to do with changes you are making in this pull request. Forget about pull request #125 for a moment and just focus on this one.

What is the work that is being done here? Changing usage of a term "rating" to "popularity". Is this directly related to the upcoming changes that you are working on to implement the rating functionality? Yes, but this is irrelevant in the context of this pull request. This is a separate part. Don't mix pull requests when they don't need to be, you'll only make it more confusing.

File: critiquebrainz/db/review.py (hunk 2 above)
Comment (ps2611) [inline critiquebrainz/db/review.py]: The changes in #125 are dependent on this PR. So its getting a bit confusing. That's the reason I asked you if I could do it in that PR as a seperate commit.

File: critiquebrainz/ws/review/views.py (hunk 3 above)
Comment (ps2611) [inline critiquebrainz/ws/review/views.py]: Okay. I will push the final changes in a bit.

File: critiquebrainz/db/review.py (hunk 2 above)
Comment (gentlecat) [inline critiquebrainz/db/review.py]: One way to keep another pull request in sync with another unmerged branch is to keep rebasing the new work on top of it. Commits from another PR will show up there, but as long as it's made clear that the work in that PR depends on another it shouldn't be a problem.

In general I'd suggest to avoid a situation like that if it's confusing to keep dependent PRs in sync.

File: critiquebrainz/db/review.py [hunk 4]
```diff
@@ -510,10 +510,10 @@ def get_popular(limit=None):
                            CASE WHEN vote = 't' THEN 1
                                 WHEN vote = 'f' THEN -1
                                 WHEN vote IS NULL THEN 0 END
-                       ) AS rating,
+                       ) AS popularity,
                        latest_revision.id AS latest_revision_id,
                        latest_revision.timestamp AS latest_revision_timestamp,
-                       latest_revision.text AS text
+                       latest_revision.text AS text,
```
Comment (gentlecat) [inline critiquebrainz/db/review.py]: Pretty sure this change is causing a syntax error.

**Comment by gentlecat** [comment]
> I finally got this. We must change the current use of 'rating' before introducing the entity rating.

Exactly!

File: critiquebrainz/db/review.py (hunk 4 above)
Comment (ps2611) [inline critiquebrainz/db/review.py]: Fixed it.
